common/security-features/subresource/subresource.py

```diff
@@ -170,20 +172,28 @@ def respond(request,
     response.writer.write_status(status_code)
 
     if access_control_allow_origin != None:
-        response.writer.write_header("access-control-allow-origin",
+        response.writer.write_header(b"access-control-allow-origin",
                                      access_control_allow_origin)
-    response.writer.write_header("content-type", content_type)
-    response.writer.write_header("cache-control", cache_control)
+    response.writer.write_header(b"content-type", content_type)
+    response.writer.write_header(b"cache-control", cache_control)
 
     additional_headers = maybe_additional_headers or {}
     for header, value in additional_headers.items():
         response.writer.write_header(header, value)
 
     response.writer.end_headers()
 
-    server_data = {"headers": json.dumps(request.headers, indent = 4)}
+    new_headers = {}
+    new_val = []
+    for key, val in request.headers.items():
+        if len(val) == 1:
+            new_headers[isomorphic_decode(key)] = isomorphic_decode(val[0])
+        else:
+            for i in val:
+                new_val.append(isomorphic_decode(i))
+            new_headers[isomorphic_decode(key)] = new_val
```

Comment:
Since new_val is never cleared, it looks like it could end up containing values from multiple headers. What if we did this:
```py
for key, val in request.headers.items():
    if len(val) == 1:
        new_val = isomorphic_decode(val[0])
    else:
        new_val = [isomorphic_decode(x) for x in val]
    new_headers[isomorphic_decode(key)] = new_val
```

Reply:
It's a much better solution. Thank you!